File: GGAndroid_2.0/GGPay/pay-core/src/main/java/io/ganguo/pay/core/PayResult.java
package io.ganguo.pay.core;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by Roger on 05/07/2017.
 */

public class PayResult implements Parcelable {
    public String type;
    public String message;
    public String code;
    public String result;

    @Override
    public String toString() {
        return "PayResult{" +
                "type='" + type + '\'' +
                ", message='" + message + '\'' +
                ", code='" + code + '\'' +
                ", result='" + result + '\'' +
                '}';
    }


    public PayResult() {
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.type);
        dest.writeString(this.message);
        dest.writeString(this.code);
        dest.writeString(this.result);
    }

    protected PayResult(Parcel in) {
        this.type = in.readString();
        this.message = in.readString();
        this.code = in.readString();
        this.result = in.readString();
    }

    public static final Creator<PayResult> CREATOR = new Creator<PayResult>() {
        @Override
        public PayResult createFromParcel(Parcel source) {
            return new PayResult(source);
        }

        @Override
        public PayResult[] newArray(int size) {
            return new PayResult[size];
        }
    };
}

File: GGAndroid_3.0/core/service/pay/pay-core/src/main/java/io/ganguo/pay/core/PayResult.kt
package io.ganguo.pay.core

import io.ganguo.pay.core.annotation.PayStatus
import io.ganguo.pay.core.annotation.PayType


/**
 * Created by Roger on 05/07/2017.
 * 支付结果回调
 * @property type 支付类型
 */
data class PayResult<T : PayOrderInfo>(
        @PayType
        @get:PayType
        @setparam:PayType
        var type: String,
        var message: String? = "",
        var code: Int = 0,
        var result: T? = null,
        @PayStatus
        @setparam:PayStatus
        @get:PayStatus
        var status: Int = PayStatus.SUCCESS
)
